src/libData/AccountData/AccountStore.cpp

```diff
@@ -501,5 +548,289 @@ void AccountStore::RevertCommitTemp() {
     RemoveFromTrie(entry.first);
   }
 
-  ContractStorage::GetContractStorage().RevertContractStates();
+  ContractStorage2::GetContractStorage().RevertContractStates();
 }
+
+bool AccountStore::MigrateContractStates() {
+  LOG_MARKER();
+
+  for (const auto& i : m_state) {
+    Address address(i.first);
+    LOG_GENERAL(INFO, "Address: " << address.hex());
+
+    Account account;
+    if (!account.DeserializeBase(bytes(i.second.begin(), i.second.end()), 0)) {
+      LOG_GENERAL(WARNING, "Account::DeserializeBase failed");
+      return false;
+    }
+    if (account.isContract()) {
+      account.SetAddress(address);
+    } else {
+      this->AddAccount(address, account);
+      continue;
+    }
+
+    std::pair<Json::Value, Json::Value> roots;
+    if (!account.GetStorageJson(roots)) {
+      LOG_GENERAL(WARNING, "Cannot get StorageJson");
+    } else {
+      LOG_GENERAL(
+          INFO, "InitJson: "
+                    << JSONUtils::GetInstance().convertJsontoStr(roots.first));
+      LOG_GENERAL(
+          INFO, "old account state: "
+                    << JSONUtils::GetInstance().convertJsontoStr(roots.second));
+    }
+
+    map<string, bytes> mutable_states;
+    // map<string, bytes> immutable_states;
+    Json::Value immutable_states;
+
+    /// generate depth_map
+
+    uint32_t scilla_version;
+    bool found_scilla_version = false;
+
+    /// retrieving immutable states (init data)
+    for (const auto& index : account.GetStorageKeyHashes()) {
+      string raw_val = account.GetRawStorage(index, false);
+
+      StateEntry entry;
+      uint32_t version;
+      if (!Messenger::GetStateData(DataConversion::StringToCharArray(raw_val),
+                                   0, entry, version)) {
+        LOG_GENERAL(WARNING, "Messenger::GetStateData failed.");
+        return false;
+      }
+
+      if (version != CONTRACT_STATE_VERSION) {
```

Comment:
Question: Are we incrementing `CONTRACT_STATE_VERSION` for v5.0.0?

Reply:
this version for the old contract storage. this data structure will be deprecated once our migration is done.